### PR TITLE
[Release] Version 0.8.1: Current_timestamp macro updates, pull request join fix

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** [#56], plus current timestamp issue reported in Databricks from a customer.

**This PR will result in the following new package version:** 0.8.1

No fields changed, just underlying logic. 

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Bug Fixes 
- Replaced the existing `dbt_current_timestamp.backcompat()` with the up-to-date `dbt_current_timestamp` macro to generate the current timestamp for customers. [PR #58](https://github.com/fivetran/dbt_github/pull/58)
- Updated the join type in `int_github__pull_request_times` to not drop pull requests without explicit reviewers requested. [PR #57](https://github.com/fivetran/dbt_github/pull/57)

## Under the Hood:
- Added consistency tests for `github__issues` and `github__pull_requests` to ensure new changes don't change the output of either model. There are conditions. [PR #58](https://github.com/fivetran/dbt_github/pull/58)

## Contributors
- [@samkessaram](https://github.com/samkessaram) ([PR #57](https://github.com/fivetran/dbt_github/pull/57))

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [NA] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [x] The appropriate issue has been linked, tagged, and properly assigned
- [x] All necessary documentation and version upgrades have been applied
- [x] docs were regenerated (unless this PR does not include any code or yml updates)
- [x] BuildKite integration tests are passing
- [x] Detailed validation steps have been provided below

### Detailed Validation
Please share any and all of your validation steps:

Validated that the new `dbt.current_timestamp` macro function in Databricks (which was where the customer reported the error) works with UTC. This was taken at 15:13 UTC time:
Screenshot 2024-10-22 at 9 12 57 AM

Validation tests were checked on development data and integration tests.

Note: Some fields in the row changes were commented out, as they relied on the current timestamp. 

Also string aggregated fields don't always have the same order, so those fields were also excluded from the row comp.

Screenshot 2024-10-22 at 8 59 17 AM
Screenshot 2024-10-22 at 8 59 04 AM

### If you had to summarize this PR in an emoji, which would it be?
 
🐛

### CODE REVIEW

File: integration_tests/tests/consistency_issues.sql
Comment (fivetran-joemarkiewicz) [inline integration_tests/tests/consistency_issues.sql:1]: Can we also add a test that explicitly checks the counts (and in the future other aggregates) of the impacted end model of this PR. For example, I would want to make sure that going forward the total row count matches between dev and prod. Something like the following would be great.
```sql
{{ config(
    tags="fivetran_validations",
    enabled=var('fivetran_validation_tests_enabled', false)
) }}

with prod as (
    select 
        1 as join_key,
        count(*) as total_prod_pr_count
    from {{ target.schema }}_github_prod.github__pull_requests   
    group by 1
),

dev as (
    select 
        1 as join_key,
        count(*) as total_dev_pr_count
    from {{ target.schema }}_github_dev.github__pull_requests  
    group by 1
),

final as (
    select
        total_prod_pr_count,
        total_dev_pr_count
    from prod
    full outer join dev
        on dev.join_key = prod.join_key
)

select *
from final
where total_dev_pr_count != total_prod_pr_count
```

As we found in the past the prod_not_in_dev and dev_not_in_prod approach doesn't always capture missing records as well as it does different records. So this added test should help close that gap!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,18 @@
+# dbt_github v0.8.1  
+[PR #58](https://github.com/fivetran/dbt_github/pull/58) contains the following updates:
+
+## Bug Fixes 
+- Replaced the existing `dbt_current_timestamp.backcompat()` with the up-to-date `dbt_current_timestamp` macro to generate the current timestamp for customers. [PR #58](https://github.com/fivetran/dbt_github/pull/58)
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:5]: Can we add a bit more to this entry explaining why this was an issue in the past and how this updated addresses that bug.

File: integration_tests/tests/consistency_issues.sql
Comment (fivetran-joemarkiewicz) [reply integration_tests/tests/consistency_issues.sql:1]: A similar aggregate test for the issue table would be great to add as well.

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-avinash these changes look good to me! I have a small suggestion to add more details to a CHANGELOG entry as well as adding two new aggregate tests for the end models impacted by these changes. I already tested the row counts locally and they passed so no need to block approval before those updates are applied.

Once all updates are applied this will be ready for release review.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,18 @@
+# dbt_github v0.8.1  
+This release contains the following updates:
+
+## Bug Fixes 
+- Replaced the existing `dbt_current_timestamp.backcompat()` with the up-to-date `dbt_current_timestamp` macro. The existing macro was not always returning the system timezone timestamp rather than the expected UTC value, causing negative downstream measures for open issues like `days_issue_open`. [PR #58](https://github.com/fivetran/dbt_github/pull/58)
+- Updated the join type in `int_github__pull_request_times` to not drop pull requests without explicit reviewers requested. [PR #57](https://github.com/fivetran/dbt_github/pull/57)
+
+## Under the Hood:
+- Added consistency tests for `github__issues` and `github__pull_requests` to ensure new changes don't change the output of either model. There are conditions. [PR #58](https://github.com/fivetran/dbt_github/pull/58)
```
[the remark this answers is not in the document]
Comment (fivetran-avinash) [reply CHANGELOG.md:9]: Oh god. lol. Fixing.

**Review by fivetran-avinash — COMMENTED** [review]
@fivetran-catfritz release review notes addressed!

**Review by fivetran-catfritz — APPROVED** [review]
Awesome! Lgtm.

**Pull request merged**